Experiments_toolkit.py

```diff
@@ -0,0 +1,190 @@
+            filtering_params.initial_filtering.divisor)
+        if len(activity_windows) > 0:
+            start_time, end_time = wl.get_start_time(
+                activity_windows, rows, filtering_params.precise_filtering.window_size,
+                filtering_params.precise_filtering.stride,
+                filtering_params.precise_filtering.threshold)
+            windowed_trace = window_function(
+                sample, start_time + offset, window_size)
+            if len(windowed_trace[0][0]) != window_size:
+                continue
+            windows.extend(windowed_trace)
+    np_windows = np.array(windows)
+    return np_windows
+
+
+def get_np_X_Y(x, y, length):
+    """
+       Description: Create X and Y numpy arrays desired length of random 
+       				samples (without replacement) from lists of windowed data, x and y
+
+       Args:
+           x -- np_array (Numpy array of one category numpy arrays with the same shape)
+           y -- np_array (Numpy array of another category numpy arrays with 
+           				  the same shape as x numpy arrays)
+           length -- int Number of samples in the numpy array to be returned 
+
+       Return:
+            X -- np array (Numpy array where the first half of the data points
+            			  come from x and second half comes from y)
+            y -- np array (Numpy array with length = length with binary labels 
+            			   (1, 0) for the differetn x, y data points)
+           """
+    len1 = len(x)
+    len2 = len(y)
+    limit = min(len1, len2, length)
+    idx = np.random.randint(len1, size=limit)
+    idx2 = np.random.randint(len2, size=limit)
+    rnd_x = x[idx]
+    rnd_y = y[idx2]
+    X = np.concatenate((rnd_x, rnd_y))
+    pos = np.ones(limit)
+    neg = np.zeros(limit)
+    y = np.concatenate((pos, neg))
+    return X, y
+
+
+def create_dataset(pos, neg, max_samples):
+    """
+       Description: Create a tf.data.Dataset composed of samples from the pos and neg list 
+
+       Args:
+           pos -- np array (Numpy arrays of positive windows, [(sensor_dims), num_samples])
+           neg -- np arrya (Numpy arrays of negative windows, [(sensor_dims), num_samples])
+           max_size -- int (number of samples to be taken from pos and neg to
+                                                be used in the dataset. If max_size < 0, then all 
+                                                samples from both pos and neg will be used) 
+
+       Return:
+           dataset -- tf.data.Dataset (TensorFlow dataset composed of the 
+                                  positive and negative lists of data passed into the 
+                                  function with the corresponding labels)
+     """
+    np.random.shuffle(pos)
+    np.random.shuffle(neg)
+    if (max_samples > 0):
+        pos = pos[:max_samples]
+        neg = neg[:max_samples]
+    examples = np.concatenate((pos, neg))
+    pos_label = np.ones(len(pos))
+    neg_label = np.zeros(len(neg))
```

Comment:
Could you just call the function above instead of repeating the operations listed here?